File: backend/api/classes/Class.py
import string
from api.classes.ClassResources import ClassResources as CR

class Class:
    __classID: int
    __className: string
    __classType: string
    __studentsEnrolled: int
    #__ClassReso: ClassResouce-- Will just declare in constructor for sake of ease
    # Default Constructor
    def __init__(self, id, name, type, numStudents):
        self.__classID = id
        self.__className = name
        self.__classType = type
        self.__studentsEnrolled = numStudents
        self.__Resource=CR.classResources()
        
    # Gets the Class Info
    def createClassResources(self, data:str):
        self.__Resource.saveClassResource(data)
        
    def getClassInfo(self) ->list:
        return self.__Resource.sendClassInfo()
    

File: frontend/classes/Class.ts



export default class Class{

  private _classId: string;
  private _className: string;
  private _classDescription?: string;


  constructor(
    classId: string,
    className: string,
    classDescription?: string
  ){
    this._classId = classId;
    this._className = className;
    this._classDescription = classDescription;
  }


  public get classId(){
    return this._classId;
  }
  public get className(){
    return this._className;
  }
  public get classDescription(){
    return this._classDescription;
  }

  public set classId(classId: string){
    this._classId = classId;
  }
  public set className(className: string){
    this._className = className;
  }
  public set classDescription(classDescription: string){
    this._classDescription = classDescription;
  }
  






}














